enterprise/api/v1/views.py

```diff
@@ -213,6 +215,110 @@ def course_enrollments(self, request, pk):
 
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
+    @action(detail=True, methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    # pylint: disable=invalid-name,unused-argument
+    def enroll_learners_in_courses(self, request, pk):
+        """
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course. This endpoint is
+        not transactional, in that any one or more failures will not affect other successful enrollments made within
+        the same request.
+
+        Expected params:
+            - licenses_info (list of dicts): an array of dictionaries, each containing the necessary information to
+            create a licenced enrollment for a user in a specified course. Each dictionary must contain a user email, a
+            course run key, and a UUID of the license that the learner is using to enroll with.
+            Example:
+                licenses_info: [
+                    {
+                        'email': 'newuser@test.com',
+                        'course_run_key': 'course-v1:edX+DemoX+Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    },
+                    ...
+                ]
+
+        Optional params:
+            - discount (int): the percent discount to be applied to all enrollments. Defaults to 100.
+
+        Expected Return Values:
+            Success cases:
+                - All users exist and are enrolled - [], 200
+                - Some or none of the users exist but are enrolled - [], 202
+
+            Failure cases:
+                - Some or all of the users can't be enrolled, no users were enrolled -
+                    {'successes': [], 'pending': [], 'failures': []}, 409
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        try:
+            serializer.is_valid(raise_exception=True)
+        except ValidationError:
```

Comment:
hmm, is catching ValidationError needed here, vs just checking the output is false? https://www.django-rest-framework.org/api-guide/serializers/#validation

Reply:
actually , hold on. Based on their doc 

```
# Return a 400 response if the data was invalid.
serializer.is_valid(raise_exception=True)
```

reads to me as if simply calling is_valid with that signature automatically take care of throwing back 400, or no? https://www.django-rest-framework.org/api-guide/serializers/#raising-an-exception-on-invalid-data

Reply:
Binod is right. It is likely just a copy paste error: there is a fair amount of code in the API that doesn't seem to leverage the Django Serializer base functionality.

Reply:
Yeah, I was trying to keep the endpoints consistent between `enterprise_learners` and `enroll_learners_in_courses` but I agree it's counter intuitive. Happy to change to proper handling.

Reply:
Oh wait, I wonder if serializer.errors doesn't get returned as the whole message if you use the built in one. I would leave this alone as well.